src/lib/apiTypesV2/Entity.cpp

```diff
@@ -212,11 +212,16 @@ std::string Entity::check(ApiVersion apiVersion, RequestType requestType)
     return std::string(errorMsg);
   }
 
-  if ((apiVersion == V2) && ((len = strlen(type.c_str())) < MIN_ID_LEN) && (requestType != BatchQueryRequest))
+
+  if (!((requestType == BatchQueryRequest) || (requestType == BatchUpdateRequest && !typeGiven)))
+
   {
-    snprintf(errorMsg, sizeof errorMsg, "entity type length: %zd, min length supported: %d", len, MIN_ID_LEN);
-    alarmMgr.badInput(clientIp, errorMsg);
-    return std::string(errorMsg);
+    if ((apiVersion == V2) && ((len = strlen(type.c_str())) < MIN_ID_LEN))
+    {
+      snprintf(errorMsg, sizeof errorMsg, "entity type length: %zd, min length supported: %d", len, MIN_ID_LEN);
+      alarmMgr.badInput(clientIp, errorMsg);
+      return std::string(errorMsg);
+    }
```

Comment:
Makes sense... I'd suggest to solve this in one of the following alternatives:

1. Just put a "FIXME Px: probably we don't need to check for V2 here as this method is only called from the NGSIv2 processing logic" (x could be "5", for instance) and left this as it is now.
2. Remove the apiVersion == V2 checking and run the .test regression to check if it is safe removing.

I prefer 1, as I don't think this deserve to wait ~1 hour to see how it goes :). However, up to you @crbrox